pennylane/data/data_manager/progress/__init__.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A library for showing loading progress, using ``rich`` or basic stdout."""
+
+from typing import Any, Optional
+
+from pennylane.data.data_manager.progress._default import (
+    make_progress as make_progress_default,
+)
+
+try:
+    from pennylane.data.data_manager.progress._rich import (
+        make_progress as make_progress_rich,
+    )
+except ImportError:  # pragma: no cover
+    make_progress_rich = None
+
+
+class Task:  # pylint: disable=too-few-public-methods
+    """Represents progress display for a single dataset download."""
+
+    def __init__(self, _task_id: Any, _progress: Any):
+        """Private constructor."""
+        self._progress = _progress
+        self._task_id = _task_id
+
+    def update(
+        self,
+        *,
+        completed: Optional[float] = None,
+        advance: Optional[float] = None,
+        total: Optional[float] = None,
+    ):
+        """Update download state.
+
+        Args:
+            advance: Adds to number of bytes downloaded so far
+            completed: Sets the number of bytes downloaded so far
+            total: Sets the total number of bytes for the download
+        """
+        self._progress.update(self._task_id, completed=completed, total=total, advance=advance)
+
+
+class Progress:
```

Comment:
Downloading to disk and loading from disk to memory both show the same progress printouts.
This is great, it's good to know when we're loading a large dataset to memory since it can sometimes take a whlie. But I'm wondering if we need to think about displaying them differently.

Reply:
It's not actually showing progress for loading into memory, just download progress. The datasets are loaded into memory after all the downloads complete